app/src/main/java/com/banuba/quickstart_c_api/OffscreenEffectPlayerImage.java

```diff
@@ -1,45 +1,125 @@
 package com.banuba.quickstart_c_api;
 
+import androidx.camera.core.ImageProxy;
+
 import java.nio.ByteBuffer;
 
 public class OffscreenEffectPlayerImage {
-    OffscreenEffectPlayerImage() {
-        mImageInfo = new ImageInfo();
+
+    OffscreenEffectPlayerImage(ImageProxy imageProxy) {
+        mWidth = imageProxy.getWidth();
+        mHeight = imageProxy.getHeight();
+        mPixelFormat = imageProxy.getImage().getFormat();
+
+        ImageProxy.PlaneProxy planeProxy = imageProxy.getPlanes()[0];
+        mRowStride0 = planeProxy.getRowStride();
+        mPlane0 = planeProxy.getBuffer();
+        mPixelStride0 = planeProxy.getPixelStride();
+        int planesNumber = imageProxy.getPlanes().length;
+        if (planesNumber > 1) {
+            planeProxy = imageProxy.getPlanes()[1];
+            mRowStride1 = planeProxy.getRowStride();
+            mPlane1 = planeProxy.getBuffer();
+            mPixelStride1 = planeProxy.getPixelStride();
+            if (planesNumber > 2) {
+                planeProxy = imageProxy.getPlanes()[2];
+                mRowStride2 = planeProxy.getRowStride();
+                mPlane2 = planeProxy.getBuffer();
+                mPixelStride2 = planeProxy.getPixelStride();
+            } else {
+                mRowStride2 = 0;
+                mPlane2 = null;
+                mPixelStride2 = 0;
+            }
+        } else {
+            mRowStride1 = 0;
+            mPlane1 = null;
+            mPixelStride1 = 0;
+            mRowStride2 = 0;
+            mPlane2 = null;
+            mPixelStride2 = 0;
+        }
     }
 
-    public ImageInfo mImageInfo;
-    public ByteBuffer mImageZero = null;
-    public ByteBuffer mImageFirst = null;
-    public ByteBuffer mImageSecond = null;
-}
+    OffscreenEffectPlayerImage(byte[] rgbPlane, int width, int height) {
+        assert rgbPlane != null;
+        mPlane0 =  ByteBuffer.wrap(rgbPlane);
+        mPlane1 =  null;
+        mPlane2 =  null;
+        mWidth = width;
+        mHeight = height;
+
+        mRowStride0 = 0;
+        mRowStride1 = 0;
+        mRowStride2 = 0;
+        mPixelStride0 = 0;
+        mPixelStride1 = 0;
+        mPixelStride2 = 0;
+        mPixelFormat = 0;
+    }
+
+    OffscreenEffectPlayerImage(byte[] yPlane, byte[] uvPlane, int width, int height) {
```

Comment:
change to:
OffscreenEffectPlayerImage(final byte[] yPlane, final int yPlaneStride, final byte[] uvPlane, final int uvPlaneStride, final int width, final int height)

Reply:
this constructor will create an Image with nv12 format